src/arch/aarch64/kernel/bootinfo.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019 Stefan Lankes, RWTH Aachen University
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use core::fmt;
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct BootInfo {
+	pub magic_number: u32,
+	pub version: u32,
+	pub base: u64,
+	pub limit: u64,
+	pub image_size: u64,
+	pub tls_start: u64,
+	pub tls_filesz: u64,
+	pub tls_memsz: u64,
+	pub current_stack_address: u64,
+	pub current_percore_address: u64,
+	pub host_logical_addr: u64,
+	pub boot_gtod: u64,
+	pub cmdline: u64,
+	pub cmdsize: u64,
+	pub cpu_freq: u32,
+	pub boot_processor: u32,
+	pub cpu_online: u32,
+	pub possible_cpus: u32,
+	pub current_boot_id: u32,
+	pub uartport: u32,
+	pub single_kernel: u8,
+	pub uhyve: u8,
+	pub hcip: [u8; 4],
+	pub hcgateway: [u8; 4],
+	pub hcmask: [u8; 4],
+}
+
+impl BootInfo {
+	pub const fn new() -> Self {
```

Comment:
Just to be a bit picky here: Isn't this rather a `default` fn?

Reply:
`default` is not `const`. ;)

Reply:
dammn :fist_raised: